src/backend/updates.rs

```diff
@@ -71,19 +78,93 @@ fn post_upgrade() {
 }
 
 async fn post_upgrade_fixtures() {
+    mutate(|state| state.read_only = true);
```

Comment:
I think we should set the `read_only` flag in the `post_upgrade` to make sure that we start in read-only mode after the upgrade without having any window of time where the flag is false.

Reply:
I agree, but this goes to back my question on the first PR. I assumed that this function is inlined. I'll wait for your response on the other PR and if it's not inline I simply remove `async` and `spawn` when calling the fixture functions.

Reply:
It runs in timer, not in post_upgrade. Note that this function cannot be inlined in post_upgrade because this function makes calls (post_upgrade is not allowed to make calls).

I was suggesting to move the sync code into post_upgrade. We could also try to do as many migration steps as possible (synchronously) inside post_upgrade. And assert that `state.failed_posts` is empty. That has a benefit that if something goes wrong in those steps, the entire upgrade will be reverted by the system, so the canister remains in a clean state.